### DISCUSSION TITLE
Feature requests

### DISCUSSION THREAD

**Discussion by holgerkrupp** [holgerkrupp/PodcastClient#13] [Announcements]
Hi,

As my favorite podcast client stopped working, I started writing my own. The plan is to create a free, open source client for iOS.

I‘m collecting requirements and feature requests here. Please check the readme for my current „official“ feature list. (https://github.com/holgerkrupp/PodcastClient/blob/main/README.md)

To keep the excitement down: I have a full time job and a family. I usually work in the evenings on my apps.

**Comment by holgerkrupp** [comment]
Wishlist from Madtodon users:

loose Requirementlist

- [ ] no backend for refreshing feeds (for cost & reliability reasons)
- [ ] inbox/queue system
- [ ] skipping chapters
- [ ] different playback speeds per feed
- [ ] Kapitelbilder
- [ ] Optional: CarPlay / Android Auto
- [ ] Playlists/Save-Queue-Function to switch between everyday/special use
- [ ] OPML Import/Export
- [ ] some users really want a good sleep timer. No idea why.
- [ ] skip forward/back 15/30s buttons in Player UI
- [x] Shownotes with clickable links
- [ ] configurable automatic download  behavior (download all new, only keep last n episodes, no auto-download) per feed

Nice to have:
- [ ] skip intro/outro per feed (n seconds)
- [ ] Chapter Images
- [ ] fyyd.de as search engine
- [ ] avoid accidentally skipping forward or backwards
- [ ] auto-skipper skipping chapters with specific keywords

**Reply by SnapeAce** [reply]
My wish list extracted from the Mastodon list

* no backend for refreshing feeds (for cost & reliability reasons)
* inbox/queue system 
* skipping chapters
* Chapter pictures
* OPML Import/Export
* skip forward/back 15/30s buttons in Player UI
* Shownotes with clickable links
* configurable automatic download behavior (download all new, only keep last n episodes, no auto-download) per feed

Nice to have:
* skip intro/outro per feed (n seconds)
* Chapter Images
* fyyd.de as search engine
* iCloud sync (If that were even possible)
* Selection of the chapters to be skipped... otherwise the skip function mentioned above

This is my wish list with the emphasis on wish

Have a great weekend and family always comes first🤗

**Reply by TheBreznsoiza** [reply]
Adding my wishlist for a podcast client (using right now PocketCasts for some years)

Must have:

- [ ] iPad version
- [ ] Notifications 
- [ ] Apperance (Light/Dark/System)
- [ ] inbox/queue system
- [ ] skipping chapters
- [ ] Chapter images
- [ ] Trim silence
- [ ] CarPlay
- [ ] OPML Import/Export
- [ ] Sleep timer (Basic, 30 Min, 1 hour, End of episode)
- [ ] skip forward/back 15/30s buttons in Player UI
- [ ] Shownotes with clickable links
- [ ] configurable automatic download behavior (download all new, only keep last n episodes, no auto-download) per feed
- [ ] Auto archive played episodes 

Nice to have:

- [ ] different playback speeds per feed
- [ ] Apple Watch 
- [ ] macOS version (maybe iPad catalysts)
- [ ] skip intro/outro per feed (n seconds)
- [ ] avoid accidentally skipping forward or backwards
- [ ] auto-skipper skipping chapters with specific keywords

**Reply by chrismarquardt** [reply]
Your proposed list of requirements looks great, let me add a few details to a few of the points:

>  inbox/queue system

preferably keep the number of taps to add an episode to a queue as low or lower than what's in the current app

>  skipping chapters

I've come to like the checkbox chapter list to de-select chapters. I love the word/pattern match idea too.

>  skip forward/back 15/30s buttons in Player UI

preferably with independently configurable skip times

>  configurable automatic download behavior (download all new, only keep last n episodes, no auto-download) per feed

would love to have a setting "auto download queued items"  

>  avoid accidentally skipping forward or backwards

YES! particularly prevent users from accidentally swiping the progress bar on the lock screen. for me this seems to be the major point of irritation

>  auto-skipper skipping chapters with specific keywords

some simple regex might be helpful (or a simple choice of location "chapter name contains | starts with | ends with")

**Comment by SnapeAce** [comment]
Hello Holger

Since we are talking about the same podcast app, I think you know exactly what the corresponding features are.

That's why I'm just leaving a thank you note.

And by the way, I have a lot of patience and know your situation, give my regards to Arne ;-)

Best regards

Marco Bieri

I'm still looking at your Wishlist from Mastodon Users…I don’t need a sleep timer.

**Comment by tobsen** [comment]
Hi Holger,

some input:

> * [ ]  skipping chapters

Would be nice to select before playing in the queue. And maybe also skip via pattern, like contains "sponsor"

> * [ ]  Optional: CarPlay / Android Auto

Useful addition would be separate setting for CarPlay, like playback speed override set to 1x.

Tobsen

**Reply by holgerkrupp** [reply]
The pre- and keyword based selection of chapters to skip is very high on my personal priority list. 😃

**Comment by thorstenstark** [comment]
Hi Holger,

great idea! I think I currently use the same client as you do. So my wished for features are mostly based on its feature set. There may be things that are already on the feature list, but I mention nevertheless to give them some more weight ;) 

- The queue system
	- auto add new episodes to top/bottom
	- ideally an option to add to very top (meaning to start with this next, even when there is currently an episode not finished, i.e. for daily news)
	- option to set audio length to pre-download i.e. load the next episodes in the queue until there are 5h to listen offline
- different playback speed per feed/podcast
- trim silence
- skip chapters (option to make this the default behavior for the skip buttons)
- per feed/podcast setting for download and archive behavior
- playback on-demand and offline
- shownotes with links and images

**Comment by mrxoom1274** [comment]
Hi Holger, a point from my wishlist : Apple Suggestion for Apple Journal App (Cover Pictures) Like as from Apple Podcast App

**Reply by holgerkrupp** [reply]
Hi. Thanks for the suggestion. From what I know and read in apples documentation, it's currently not possible to provide suggestions to the journal app, it's only possible to receive Suggestions if someone is building their own Journal App. 
https://developer.apple.com/videos/play/tech-talks/111384/
https://developer.apple.com/documentation/journalingsuggestions

But I will keep this in mind, as I like the idea to implement something like this in the future. Thanks
